### DISCUSSION TITLE
Mysql 查询逗号分隔符的数组字段

### DISCUSSION THREAD

**Discussion by lhtuling** [dotnetcore/FreeSql#1016] [Q&A]
这种字段要查询是否包含怎么查询更好！  用   like '%,1,%'  这样的  第一个和最后一个会匹配不到  不准确  有没有更好的方式！ 
![image](https://user-images.githubusercontent.com/44140500/150094376-0d8ed44d-24be-490f-8b8d-53560f40ec09.png)

**Comment by lhtuling** [comment]
想到一个方式。。。暂时先用吧 希望大佬能提供个更好的方式  
.Where(it => $",{it.ToUsers},".Contains($",{userid},"));

**Reply by luoyunchong** [reply]
存数据时候就存成     ,'12','13',

**Reply by lhtuling** [reply]
好吧。。。那还是一样效果  就在想有木有其他不知道的更方便的方式  没有的话就这样也能用~~~

**Reply by hmwz009** [reply]
我本来还想是否直接可以存成List类型，然后直接用Contains进行查询。（JsonMap）

**Comment by hd2y** [comment]
另外一个解题思路，用正则，不过无论是模糊检索或者正则匹配，效率应该都不怎么样。

相关内容参考：https://dev.mysql.com/doc/refman/8.0/en/regexp.html
